src/handler.py

```diff
@@ -287,9 +312,16 @@ def on_successful_auth(cli, blah, blahh, blahhh):
         @hook("906")
         @hook("907")
         def on_failure_auth(cli, *etc):
-            alog("Authentication failed.  Did you fill the account name "
-                 "in botconfig.USERNAME if it's different from the bot nick?")
-            cli.quit()
+            nonlocal supported_sasl, selected_sasl
```

Comment:
Here, `supported_sasl` doesn't need to be non-local (although `selected_sasl` does).